### PR TITLE
feat: Add tokenSortConfig to preferences controller

### PR DESCRIPTION
**Pull request by gambinish**
## Explanation

Adds a `tokenSortConfig` to `PreferencesController` to pair with mobile and extension codebases. This config will tell each application which keys tokens are sorted by, which order they are sorted in, and by which sorting callback they should use while being sorted.

## References

https://consensyssoftware.atlassian.net/browse/MMASSETS-357
https://consensyssoftware.atlassian.net/browse/MMASSETS-356

## Changelog

<!--
If you're making any consumer-facing changes, list those changes here as if you were updating a changelog, using the template below as a guide.

ADDED tokenSortConfig to PreferencesController default state

## Checklist

- [x] I've updated the test suite for new or updated code as appropriate
- [x] I've updated documentation (JSDoc, Markdown, etc.) for new or updated code as appropriate
- [x] I've highlighted breaking changes using the "BREAKING" category above as appropriate
- [x] I've prepared draft pull requests for clients and consumer packages to resolve any breaking changes

### CODE REVIEW

**Comment by sahar-fehri** [comment]
Looks good! 
Can we update change log to reflect the update; sth like

### `@metamask/preferences-controller`

- **Added**: Added `tokenSortConfig` to preferencesController state

File: packages/preferences-controller/src/PreferencesController.ts [hunk 1]
```diff
@@ -199,6 +200,11 @@ export function getDefaultPreferencesState() {
     useTokenDetection: true,
     smartTransactionsOptInStatus: false,
     useTransactionSimulations: true,
+    tokenSortConfig: {
```
Comment (sahar-fehri) [inline packages/preferences-controller/src/PreferencesController.ts:203]: Should we add `tokenSortConfig` in type `PreferencesState` ?

Comment (gambinish) [reply packages/preferences-controller/src/PreferencesController.ts:203]: Added 👍

**Comment by sahar-fehri** [comment]
Maybe we wont this one if this gets merged first https://github.com/MetaMask/core/pull/4860/files#diff-40a12640df782cc92dd07efa2b2f120024eb7c5cf1d9431893406ef39132e272

**Comment by cryptodev-2s** [comment]
Closes as change was included in https://github.com/MetaMask/core/pull/4860

**Pull request closed without merging**